### PR TITLE
GH871 alternative patch

### PR DESCRIPTION
**Pull request by slfritchie**
This is an alternative to the patch to #871.  I didn't have a complete understanding of the circumstances that #871 is supposed to fix, so I made my best guess.  If I'm wrong, then this PR should be closed.

My understanding was that @Maithem was running into a case where `fillReadQueue` would encounter at `TrimmedException` before a checkpoint `START` record was found, and then the map state was corrupted because the checkpoint's data was not discarded.  There are a few cases where a `TrimmedException` can be thrown by runtime without seeing the exception from the server; see AbstractQueuedStreamView.java.

This PR will forcibly clear the stream context's `readCpQueue` if `START` is not seen.

### CODE REVIEW

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java [hunk 1]
```diff
@@ -90,7 +90,26 @@ public void testEmptyMapCP() throws Exception {
...
+        boolean[] b = new boolean[]{true, false};
+        int counter = 0;
+
+        for (int i = 0; i < 2; i++) {
+            for (int j = 0; j < 2; j++) {
+                for (int k = 0; k < 2; k++) {
+                    // Avoid the case of writing START & END but not the
+                    // checkpoint data in the middle CONTINUATION (index=j).
+                    if (! (b[i] && !b[j] && b[k])) {
+                        System.err.printf("try %s %s %s %d\n", b[i], b[j], b[k], counter);
```
Comment (no2chem) [inline test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java]: System.err doesn't belong in a test

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java [hunk 2]
```diff
@@ -90,7 +90,26 @@ public void testEmptyMapCP() throws Exception {
      */
     @Test
 	public void smoke1Test() throws Exception {
```
Comment (no2chem) [inline test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java]: smoke1test is not a good name for a test

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java [hunk 3]
```diff
@@ -90,7 +90,26 @@ public void testEmptyMapCP() throws Exception {
...
-        final String streamName = "mystream";
+        boolean[] b = new boolean[]{true, false};
+        int counter = 0;
+
+        for (int i = 0; i < 2; i++) {
+            for (int j = 0; j < 2; j++) {
+                for (int k = 0; k < 2; k++) {
+                    // Avoid the case of writing START & END but not the
+                    // checkpoint data in the middle CONTINUATION (index=j).
+                    if (! (b[i] && !b[j] && b[k])) {
```
Comment (no2chem) [inline test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java]: It's pretty hard to determine what this test does - what is i, j and k?

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **3742ed7**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #876 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/TNK4yknF2FWBPLlWUkx97ckC1O4eqqaT)

**Comment by slfritchie** [comment]
Addressed by #889

**Pull request closed without merging**